CMakeLists.txt

```diff
@@ -16,12 +16,40 @@ ExternalProject_Add(googletest
   LOG_BUILD         ON
 )
```

Comment:
Let's also move `ExternalProject_Add(googletest` and related stuff to `cmake/download_googletest`, by analogy with `cmake/download_roc`?

Reply:
Done